### PR TITLE
Feature/drag and drop table

### PR DESCRIPTION
**Pull request by timloudon**
Closes issue #113

### CODE REVIEW

**Comment by timloudon** [comment]
Not exactly sure why the package-lock.json file is in there, nor why it is all edited.

Also, I think I could probably structure this feature more clearly although I'm not sure what the best approach would be. Comments welcome.

File: quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/QuestionCreator.tsx [hunk 1]
```diff
@@ -106,6 +50,78 @@ export default function QuestionCreator(props: any) {
...
+  const onDragEnd = (res: any) => {
+    const { destination, source } = res;
+    // if dropped outside table:
+    if (!destination) return;
+    // if dropped back in same position:
+    if (
+      destination.droppableId === source.droppableId &&
+      destination.index === source.index
+    )
+      return;
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/QuestionCreator.tsx:62]: Is this indentation right?

File: quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/QuestionCreator.tsx [hunk 2]
```diff
@@ -144,59 +156,40 @@ export default function QuestionCreator(props: any) {
...
+                  container
+                  spacing={1}
+                  ref={provided.innerRef}
+                  {...provided.droppableProps}
+                >
+                  {dataState.data.map((x: Row, i: number) => (
+                    <Question
+                      key={x.number}
+                      x={x}
+                      i={i}
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/QuestionCreator.tsx:172]: You might want to give these better names.

File: quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/Question.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  cancelEdit: any;
+  handleChange: any;
+}) {
+  const classes = useStyles();
+
+  const editingRow = props.currentlyEditing && props.editIndex === props.i;
+  const deletingRow = props.currentlyDeleting && props.deleteIndex === props.i;
+  return (
+    <Grid container item xs={12}>
+      <Draggable draggableId={props.x.number.toString()} index={props.i}>
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuizCreating/QuestionCreator/Question.tsx:66]: You might want to split this into multiple components.

**Review by calumtomeny — CHANGES_REQUESTED** [review]
Great addition to the app!

**Review by calumtomeny — APPROVED** [review]

**Pull request merged**